src/linalg.jl

```diff
@@ -227,28 +234,28 @@ end
     end
 end
 
-_norm_p0(x) = x == 0 ? zero(x) : one(x)
+_norm_p0(x) = iszero(x) ? zero(float(_inner_eltype(x))) : one(float(_inner_eltype(x)))
```

Comment:
Shouldn't there be a `float(real(...` to handle complex numbers right?

Reply:
Nice catch - definitely..!